tests/llm/test_llm_functions.py

```diff
@@ -93,3 +99,139 @@ def test_assistant_functions():
             raise ValueError(f"Function not found in test: {name_}")
 
         assert_parameters(method_definition, llm_function_dict_)
+
+
+def test_get_annotation_from_store_by_feature_list():
+    """Test that the function retrieves the correct entry from annotation store."""
+    annotation_store = {
+        "feature1": {"annotation": "annotation1"},
+        "feature3": {"annotation": "annotation3"},
+        "feature2": {"annotation": "annotation2"},
+    }
+
+    features = ["feature1"]
+    result = get_annotation_from_store_by_feature_list(features, annotation_store)
+    assert result == {"annotation": "annotation1"}
+
+    features = ["feature2", "feature3"]
+    result = get_annotation_from_store_by_feature_list(features, annotation_store)
+    assert result == {"annotation": "annotation2"}
+
+    features = []
+    result = get_annotation_from_store_by_feature_list(features, annotation_store)
+    assert result is None
+
+    features = ["feature4"]
+    result = get_annotation_from_store_by_feature_list(features, annotation_store)
+    assert result is None
+
+
+@patch("alphastats.llm.llm_functions.get_annotations_for_feature")
+@patch("streamlit.session_state")
+def test_get_annotation_from_uniprot_by_feature_list(
+    mock_session_state, mock_uniprot_annotation
+):
+    """Test that the function retrieves the correct entry from UniProt."""
+
+    mock_session_state.return_value = {StateKeys.ANNOTATION_STORE: {}}
+
+    # just one feature
+    features = ["id1"]
+    mock_uniprot_annotation.return_value = "id1"
+    result = get_annotation_from_uniprot_by_feature_list(features)
+    mock_uniprot_annotation.assert_called_with("id1")
+    mock_session_state.__getitem__().__setitem__.assert_called_with("id1", "id1")
+    assert result == "id1"
+
+    # same base ids
+    features = ["id2", "id2;id2-3"]
+    get_annotation_from_uniprot_by_feature_list(features)
+    mock_uniprot_annotation.assert_called_with("id2")
+
+    # different base ids, no longest element
+    features = ["id2", "id3"]
+    get_annotation_from_uniprot_by_feature_list(features)
+    mock_uniprot_annotation.assert_called_with("id2")
+
+    # different base ids, longest element
+    features = ["id2", "id2;id3"]
+    get_annotation_from_uniprot_by_feature_list(features)
+    mock_uniprot_annotation.assert_called_with("id2;id3")
```

Comment:
is there a reason why you are not asserting the return value?

Reply:
The mock call is the last call in the function and I asserted once in the first test, that its return value (mocked) is the final return value of the tested function. The tests aim at the value the mocked function is called with.